subworkflows/nf-core/fastq_align_bamcmp_bwa/main.nf

```diff
@@ -0,0 +1,55 @@
+//
+// Alignment with BWA to an additional genome, then use BAMCMP to remove reads that map to the second genome.
+//
+
+include { BWA_MEM as BWA_MEM_PRIMARY     } from '../../../modules/nf-core/bwa/mem/main'
+include { BWA_MEM as BWA_MEM_CONTAMINANT } from '../../../modules/nf-core/bwa/mem/main'
+include { BAMCMP                         } from '../../../modules/nf-core/bamcmp/main'
+include { BAM_SORT_STATS_SAMTOOLS        } from '../bam_sort_stats_samtools/main'
+
+workflow FASTQ_ALIGN_BAMCMP_BWA {
+    take:
+    ch_reads             // channel (mandatory): [ val(meta), [ path(reads) ] ]
+    ch_primary_index     // channel (mandatory): [ val(meta2), path(index) ]
+    ch_contaminant_index // channel (mandatory): [ val(meta3), path(ch_contaminant_index) ]
+    val_sort_bam         // boolean (mandatory): true or false
+    ch_fasta             // channel (optional) : [ path(fasta) ]
```

Comment:
```suggestion
    ch_fasta             // channel (optional) : [ val(meta4), path(fasta) ]
```
I know it doesn't make sense now, but this will have to be changed in the future eventually so why not now :grin: